weed/shell/command_volume_server_evacuate.go

```diff
@@ -80,71 +90,84 @@ func (c *commandVolumeServerEvacuate) volumeServerEvacuate(commandEnv *CommandEn
 
 	// list all the volumes
 	// collect topology information
-	topologyInfo, _, err := collectTopologyInfo(commandEnv, 0)
+	c.topologyInfo, _, err = collectTopologyInfo(commandEnv, 0)
 	if err != nil {
 		return err
 	}
 
-	if err := c.evacuateNormalVolumes(commandEnv, topologyInfo, volumeServer, skipNonMoveable, applyChange, writer); err != nil {
+	go func() {
+		for {
+			if topologyInfo, _, err := collectTopologyInfo(commandEnv, 5*time.Minute); err != nil {
```

Comment:
This is not a good design. 

* The info would not be deterministic. It is also hard to debug.
* Each time this command is run, a new goroutine will start and run forever until the `weed shell` exits.

Reply:
> This is not a good design.

@chrislusf  Would it be better if the topology update happens synchronously for the next volum if the evacuation lasted more than 5 minutes?

Reply:
updating synchronously is more deterministic.